stream_alert_cli/test.py

```diff
@@ -361,11 +376,42 @@ def check_keys(self, test_event):
             record_keys.difference_update(key_diff)
             self.status_messages.append(StatusMessage(StatusMessage.WARNING, message))
 
-        return record_keys.issubset(required_keys | optional_keys)
+        return record_keys.issubset(required_keys | optional_keys | input_data_keys)
+
+    def apply_template(self, test_event):
+        """Apply default values to the given test event
+
+        Args:
+            test_event (dict): The loaded test event
+        """
+        event_log = self.cli_config['logs'].get(test_event['log'])
+
+        parser = event_log['parser']
+        schema = event_log['schema']
+        configuration = event_log.get('configuration', {})
+
+        # Add envelope keys
+        if configuration.get('envelope_keys'):
+            schema.update(configuration['envelope_keys'])
+
+        # Setup the parser to access default optional values
+        self.parsers[parser] = self.parsers.get('parser', get_parser(parser))
+
+        # Setup the modified test event
+        default_test_event = {}
+        data = test_event['override_record'].copy()
+
+        # Add apply default values based on the declared schema
+        for key, value in schema.iteritems():
+            default_test_event[key] = self.parsers[parser].default_optional_values(value)
```

Comment:
since `default_optional_values` is now a static method on `ParserBase` you can make this an easy comprehension using:

`default_test_event = {key: ParserBase.default_optional_values(value) for key, value in schema.iteritems()`